android/src/main/kotlin/com/nt4f04und/android_content_provider/AndroidContentProvider.kt

```diff
@@ -90,26 +92,13 @@ abstract class AndroidContentProvider : ContentProvider(), LifecycleOwner, Utils
      */
     abstract val entrypointName: String // TODO: eventually replace it with initialArguments
 
-    override fun getLifecycle(): Lifecycle {
-        ensureLifecycleInitialized()
-        return lifecycle!!
-    }
-
-    private fun ensureLifecycleInitialized() {
-        if (lifecycle == null) {
-            lifecycle = LifecycleRegistry(this)
-        }
-    }
-
     @CallSuper
     override fun onCreate(): Boolean {
         val flutterLoader = FlutterInjector.instance().flutterLoader()
         flutterLoader.startInitialization(context!!.applicationContext)
         val entrypoint = DartExecutor.DartEntrypoint(flutterLoader.findAppBundlePath(), entrypointName)
         val engineGroup = getFlutterEngineGroup(context!!)
         engine = engineGroup.createAndRunEngine(context!!, entrypoint)
-        ensureLifecycleInitialized()
-        lifecycle!!.handleLifecycleEvent(Lifecycle.Event.ON_CREATE)

```

Comment:
This still exists, but you need to keep the lifecycle and lifecycle registry variables separately in order to be able to access this method. Removing it is not good

Reply:
Thanks that makes sense.

Reply:
Taking a closer look I'm not sure what the right thing to do is here. It looks lke `handleLifecycleEvent` has been removed and no longer exists in the updated `LifecyleRegistry`:

https://developer.android.com/reference/androidx/lifecycle/LifecycleRegistry

From my reading of the docs it looks like when we do this:

`LifecycleRegistry(this)`

that means it will automatically call our `onCreate` right?

I'm not very good with Android or Kotlin though so maybe I'm missing something?

Reply:
Sorry nevermind I think I figured it out.